articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,297 @@
+---
+title: Login
+description: This tutorial demonstrates how to add user login to a React application using Auth0.
```

Comment:
All examples in this quickstart use hooks. We also need to provide examples when using classes.

Reply:
@sandrinodimattia @stevehobbsdev My proposed solution to this would be to have a QuickStart based on function components with Hooks and another based on class components and the `context` object. That's exactly what I plan to do for the blog posts. 

I could even go further and suggest that we need a dedicated QS that uses TypeScript -- it's seeing tremendous growth in the React community and production apps.

Reply:
I'm not sure a second quickstart is the answer here, but what would the quickstart doc look like if we did use TypeScript? It would look 99% the same, as many of the code samples in the doc would not need changing as their types can be inferred.

The actual sample, however, is currently being converted to use TypeScript by default.

> My proposed solution to this would be to have a QuickStart based on function components with Hooks and another based on class components and the context object.

We have docs and examples as to how our hook can be used with class components - we should just link to those and default to hooks for the quickstart. As you mentioned earlier, the nuances with classes vs. hooks could be explained in more detail in a deep-dive blog post.

Reply:
![image](https://user-images.githubusercontent.com/655409/83081363-cb94b500-a080-11ea-8521-bf3a570aee91.png)

Can we have a picker of some sorts? The quickstart will always be the same, but you should be able to choose between Hooks, Hooks with TS, Classes, Classes with TS

Reply:
@sandrinodimattia I agree this would be really awesome. [We do something like this on some of our docs](https://auth0.com/docs/api-auth/tutorials/authorization-code-grant-pkce) ([here is the source](https://github.com/auth0/docs/blob/master/articles/api-auth/tutorials/authorization-code-grant-pkce.md) as to how this is done).

@dan-auth0, can you explore implementing this for one of the snippets and see how you get on?

Reply:
Addressed in CF.